### PR TITLE
[deploy_bmh] Add rootDeviceHints field support

### PR DESCRIPTION
**Pull request by pablintino**
As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

### CODE REVIEW

File: roles/deploy_bmh/template/bmh.yml.j2 [hunk 1]
```diff
@@ -20,9 +20,8 @@ spec:
   bootMACAddress: {{ nic.mac }}
 {% endfor                                                                            %}
   bootMode: {{ node_data['boot_mode'] }}
-  online:
-{%- if node_data['status'] == "running"                                              %}
- true
-{% else                                                                              %}
- false
+  online: {{ 'true' if node_data['status'] == "running" else 'false' }}
```
Comment (rebtoor) [inline roles/deploy_bmh/template/bmh.yml.j2:23]: <3

**Review by rebtoor — APPROVED** [review]

File: roles/deploy_bmh/template/bmh.yml.j2 (hunk 1 above)
Comment (cjeanner) [reply roles/deploy_bmh/template/bmh.yml.j2:23]: ah, you're using that data from the vbmc output? Good to know it's useful :). We'll need the same from sushy I guess?

Comment (pablintino) [reply roles/deploy_bmh/template/bmh.yml.j2:23]: @abays Good to know, I'll amend this by allowing `status` to not be passed and simply default it to false. For real BM we won't pass the value and that will be enough.

**Review by rebtoor — APPROVED** [review]

**Pull request merged**